docs/k8s/run.sh

```diff
@@ -89,5 +89,22 @@ if [[ ! -z "${TS_DEST_IP}" ]]; then
   iptables -t nat -I PREROUTING -d "$(tailscale --socket=${TS_SOCKET} ip -4)" -j DNAT --to-destination "${TS_DEST_IP}"
 fi
 
+if [[ ! -z "${TS_KUBE_SECRET}" ]]; then
+	DEVICE_ID=$(tailscale --socket="${TS_SOCKET}" status --json | jq -r .Self.ID)
+	echo "Device ID is $DEVICE_ID"
+	NS=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+	TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+	if ! curl -s -X PATCH \
+		 --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+		 -H "Authorization: Bearer $TOKEN" \
+		 -H "Accept: application/json" \
+		 -H "Content-Type: application/strategic-merge-patch+json" \
+		 --data-raw "{\"metadata\":{\"annotations\":{\"tailscale.com/device-id\": \"$DEVICE_ID\"}}}" \
+		 "https://kubernetes/api/v1/namespaces/${NS}/secrets/${TS_KUBE_SECRET}?fieldManager=kubectl-patch" >/dev/null; then
```

Comment:
to be safe you want to use `https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT_HTTPS}`

https://kubernetes.io/docs/concepts/services-networking/connect-applications-service/#environment-variables

Reply:
Done